### DISCUSSION TITLE
CREATED A VIRTUAL TABLE AND WANT TO ADDCOLUMNS

### DISCUSSION THREAD

**Discussion by GT2010** [DaxStudio/DaxStudio#982] [Q&A]
I need to know syntax
![2022-11-01 11-59-04 AM](https://user-images.githubusercontent.com/116607803/199305276-3991ab55-538c-4a66-b9af-7ebc1c4ee160.png)

**Comment by dgosbell** [comment]
That is not a valid syntax for the ADDCOLUMNS function (see https://learn.microsoft.com/en-us/dax/addcolumns-function-dax ) It expects a table as the first parameter, so the easiest fix is to just wrap your `[CatKey]` column reference in the VALUES() function.

eg.
```
EVALUATE
ADDCOLUMNS(
VALUES( 'MATERIAL STATUS'[CatKey] ),
"@orderQty", [Demand Qty],
"@poQty", [Total Order Qty],
"@shipQty" , [Total ship Qty],
"@recQty" , [Total Received Qty],
"@mirQty" , [Issue Qty]
 ) 
```

**Comment by GT2010** [comment]
THANKS
